### ISSUE TITLE
Fix Autonomous not running

### ISSUE THREAD

**Issue by dandeduck** [Flash3388/FlashFRC#6]

**Comment by tomtzook** [comment]
We don't actually know for sure the FMS is related. It doesn't sound connected.

**Comment by dandeduck** [comment]
It's probably not, but I don't recall ever running autonomous stuff with the `autonomous()` mode, so I think the problem is with it. At any rate, it doesn't look like it's my fault, at least from a glance and from what I remember.

**Comment by dandeduck** [comment]
I suspect that the something cancels the action right when it's started by cancelling the running action on the system or something... I know for sure that `autonomousPeriodic()` did run. The action was started in `autonomousInit`. Also, I think that's the commit related to the issues: https://github.com/Flash3388/Flash2020/commit/c69fc9f0749ab0a8921d10632a7e01bfc48f4a95
hail my naming!

**Comment by tomtzook** [comment]
> Also, I think that's the commit related to the issues: Flash3388/Flash2020@c69fc9f

You mean caused or fixed?

>  I know for sure that autonomousPeriodic() did run

Did you see it run once? Or actually periodically?

**Comment by dandeduck** [comment]
> You mean caused or fixed?

The commit is called "hopefully fixed Tom's shit "

> Did you see it run once? Or actually periodically?

I don't know, I'm just sure that what I put their worked.

I looked at the actual changes, and turns out it's related to another issue where `Actions.parallel()`didn't work, and I had to create my own crude implementation. I can't remember if this was brought up to you...

**Comment by tomtzook** [comment]
> I looked at the actual changes, and turns out it's related to another issue where Actions.parallel()didn't work, and I had to create my own crude implementation. I can't remember if this was brought up to you...

Unrelated to this issue or maybe to cause? Hard for me to say because I haven't seen any data from runs.

**Comment by dandeduck** [comment]
It's unrelated to both, what?

**Comment by tomtzook** [comment]
You mentioned `Actions.parallel` here. I assumed you suggested a connection to this problem....

Because it might be related. Did this commit fix the autonomous as well?

**Comment by dandeduck** [comment]
No, this commit was done before the tournament, but the issue was discovered during. What I did to fix is just:
```
if(!auto.isRunning())
     auto.start()
```
In `autonomousPeriodic`

**Comment by tomtzook** [comment]
Did the systems use `setDefaultAction`? Or at least one of them?

**Comment by dandeduck** [comment]
> Did the systems use `setDefaultAction`? Or at least one of them?

Yes, one of them definitely did.

**Comment by tomtzook** [comment]
The 2.X `SingleThreadScheduler` iteration unintentionally gave priority to _default_ actions at the beginning of new control modes. This might have been the cause. It was fixed already.

We'll have to test to make sure of course.
